pkgs/development/libraries/openssl/default.nix

```diff
@@ -192,15 +192,15 @@ in {
   };
 
   openssl_1_1 = common {
-    version = "1.1.1l";
-    sha256 = "sha256-C3o+XlnDSCf+DDp0t+yLrvMCuY+oAIjX+RU6oW+na9E=";
+    version = "1.1.1m";
+    sha256 = "sha256-+JGZvosjykX8fLnx2NPuZzEjGChq0DD1MWrKZGLbbJY=";
     patches = [
       ./1.1/nix-ssl-cert-file.patch
 
       (if stdenv.hostPlatform.isDarwin
        then ./use-etc-ssl-certs-darwin.patch
        else ./use-etc-ssl-certs.patch)
-    ] ++ lib.optionals (stdenv.isDarwin) [
+    ] ++ lib.optionals (stdenv.isDarwin && (builtins.substring 5 5 version) < "m") [
```

Comment:
@risicle PTAL.

Reply:
Early-stage openssl certainly builds now. Still building the full loop though.

Reply:
This doesn't take effect without `rec` after `common`.

```
applying patch /nix/store/zj80b7imvm8s8dsnnnnyyyxmm7bfk0z4-macos-yosemite-compat.patch
patching file crypto/rand/rand_unix.c
Reversed (or previously applied) patch detected!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
```

Just out of curiosity, what does the `version` here without `rec` refer to?

Reply:
It refer to `lib.version`. `with lib;` (or `with` itself) considered harmful.

Reply:
Created a PR #155079.